Tools/MSBuildTaskExtractChangelog/ExtractChangelog.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace MSBuildReadTask
+{
+    public class ExtractChangelog : Task
+    {
+        [Required]
+        public string FilePath { get; set; }
+
+        [Output]
+        public string ExtractedText { get; private set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                var textToParse = File.ReadAllText(FilePath);
+                var regex = new Regex("(?sm)^(## \\d{1,2}\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})? \\(\\d{4}-\\d{2}-\\d{2}\\))(.+?)(?=\n## \\d{1,2}\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})? \\(\\d{4}-\\d{2}-\\d{2}\\))");
+                var matches = regex.Matches(textToParse);
+                ExtractedText = matches[0].Groups[0].Value;
+                File.WriteAllText("./ExtractedChangelog.md", ExtractedText);
```

Comment:
Instead of writing to the file directly from the MSBuild task, a better approach would be to use another task to do that. You can use [WriteLinesToFile](https://docs.microsoft.com/en-us/visualstudio/msbuild/writelinestofile-task?view=vs-2019) in the Realm.csproj to stream the changelog contents to a file.

Reply:
The reason for this being that it's better to keep the task as focused as possible? Or is there something else?